credentialbuilder/src/main/java/org/apache/ranger/credentialapi/CredentialReader.java

```diff
@@ -17,80 +17,82 @@
 
-public class CredentialReader {
+import java.util.ArrayList;
+import java.util.List;
 
-  public static String getDecryptedString(String CrendentialProviderPath,String alias, String storeType) {
-	  String credential=null;
-	  try{
-		  if(CrendentialProviderPath==null || alias==null){
-			  return null;
-		  }		  		
-		  char[] pass = null;
-		  Configuration conf = new Configuration();
-		  String crendentialProviderPrefixJceks=JavaKeyStoreProvider.SCHEME_NAME + "://file";
-		  String crendentialProviderPrefixLocalJceks="localjceks://file";
-		  crendentialProviderPrefixJceks=crendentialProviderPrefixJceks.toLowerCase();
+public class CredentialReader {
+    private CredentialReader() {
+    }
 
-		  String crendentialProviderPrefixBcfks = "bcfks" + "://file";
-		  String crendentialProviderPrefixLocalBcfks= "localbcfks" + "://file";
-		  crendentialProviderPrefixBcfks=crendentialProviderPrefixBcfks.toLowerCase();
-		  crendentialProviderPrefixLocalBcfks=crendentialProviderPrefixLocalBcfks.toLowerCase();
+    public static String getDecryptedString(String credentialProviderPath, String alias, String storeType) {
+        String credential = null;
+        try {
+            if (credentialProviderPath == null || alias == null) {
+                return null;
+            }
+            char[] pass = null;
+            Configuration conf = new Configuration();
+            String crendentialProviderPrefixJceks = JavaKeyStoreProvider.SCHEME_NAME + "://file";
+            String crendentialProviderPrefixLocalJceks = "localjceks://file";
+            crendentialProviderPrefixJceks = crendentialProviderPrefixJceks.toLowerCase();
 
-		  CrendentialProviderPath=CrendentialProviderPath.trim();
-		  alias=alias.trim();
-		  if(CrendentialProviderPath.toLowerCase().startsWith(crendentialProviderPrefixJceks) ||
-				  CrendentialProviderPath.toLowerCase().startsWith(crendentialProviderPrefixLocalJceks) ||
-				  CrendentialProviderPath.toLowerCase().startsWith(crendentialProviderPrefixBcfks) ||
-				  CrendentialProviderPath.toLowerCase().startsWith(crendentialProviderPrefixLocalBcfks)){
-			  conf.set(CredentialProviderFactory.CREDENTIAL_PROVIDER_PATH,
-					   //UserProvider.SCHEME_NAME + ":///," +
-			  CrendentialProviderPath);
-		  }else{
-			  if(CrendentialProviderPath.startsWith("/")){
-			  	 if(StringUtils.equalsIgnoreCase(storeType, "bcfks")) {
-					 conf.set(CredentialProviderFactory.CREDENTIAL_PROVIDER_PATH, CrendentialProviderPath);
-				 } else {
-					 conf.set(CredentialProviderFactory.CREDENTIAL_PROVIDER_PATH,
-							 //UserProvider.SCHEME_NAME + ":///," +
-							 JavaKeyStoreProvider.SCHEME_NAME + "://file" + CrendentialProviderPath);
-				 }
+            String crendentialProviderPrefixBcfks = "bcfks" + "://file";
+            String crendentialProviderPrefixLocalBcfks = "localbcfks" + "://file";
+            crendentialProviderPrefixBcfks = crendentialProviderPrefixBcfks.toLowerCase();
```

Comment:
To improve readability, I suggest separating variable declarations, assignments and rest of the code.

Reply:
Can this be included in checkstyle rule itself, as a part of this exercise it can be done but will be difficult to maintain this restriction in future.

Reply:
AFAIK, there is no checkstyle rule available for this. if/when it becomes available, we should include it in our checkstyle.xml.